macros/Compat.ejs

```diff
@@ -1,66 +1,436 @@
+    output += `${flagText} compile flag${valueToSet}.`;
+    }
+
+  return output;
 }
 
-if (output === '') {
-  output = noDataFoundl10n;
+
+/*
+Main function responsible for the contents of a support cell in the table.
+
+`supportData` is a support_statement
+`browserId` is a compat_block browser ID
+`compatNotes` is collected Compatibility notes
+
+*/
+function writeSupportInfo(supportData, browserId, compatNotes) {
+  let output = '';
+
+  // browsers are optional in the data, display them as "?" in our table
+  if (!supportData) {
+    output = getVersionString(null);
+  // we have support data, lets go
+  } else {
+    output += getVersionString(supportData.version_added);
+
+    if (supportData.version_removed) {
+      // We don't know when
+      if (typeof(supportData.version_removed) === 'boolean' && supportData.version_removed) {
+        output += '&nbsp;—?'
+      } else { // We know when
+        output += '&nbsp;— ' + supportData.version_removed;
+      }
+    }
+
+    // Add prefix
+    if (supportData.prefix) {
+      output += `<span title="prefix" class="inlineIndicator prefixBox prefixBoxInline">
+                      <a title="The name of this feature is prefixed with '${supportData.prefix}' as this
+                      browser considers it experimental" href="/en-US/docs/Web/Guide/Prefixes">${supportData.prefix}
+                      </a></span>`;
+    }
+
+    // Add alternative name
+    if (supportData.alternative_name) {
+      output += ` (as <code>${supportData.alternative_name}</code>)`;
+    }
+
+    // Add note anchors
+    // There are two types of notes (notes, and flag notes).
+    // Collect them and order them, before adding them to the cell
+    let noteAnchors = [];
+
+    // Generate notes, if any
+    if (compatNotes && supportData.notes) {
+      if (Array.isArray(supportData.notes)) {
+        for (let note of supportData.notes) {
+          let noteIndex = compatNotes.indexOf(note);
+          noteAnchors.push(`<sup><a href="#compatNote_${noteIndex+1}">${noteIndex+1}</a></sup>`);
+        }
+      } else {
+        let noteIndex = compatNotes.indexOf(supportData.notes);
+        noteAnchors.push(`<sup><a href="#compatNote_${noteIndex+1}">${noteIndex+1}</a></sup>`);
+      }
+    }
+
+    // there is a flag and it needs a note, too
+    if (compatNotes && supportData.flag) {
+      let flagNote = writeFlagsNote(supportData, browserId);
+      let noteIndex = compatNotes.indexOf(flagNote);
+      noteAnchors.push(`<sup><a href="#compatNote_${noteIndex+1}">${noteIndex+1}</a></sup>`);
+    }
+    noteAnchors = noteAnchors.sort();
+    output += noteAnchors.join(' ');
+
+  }
+  return output;
+}
+
+/*
+Generate an aggregate table that summarises the support status for the
+individual features under the data it's passed.
+Each row in the table will contain the "basic_support" data for a single feature.
+The first cell in each the row will contain a link to the feature's page.
```

Comment:
Sorry, I should have caught this before. Also a nit, but in this version the feature name is not linked.